src/main/java/org/gridsuite/loadflow/server/service/LoadFlowWorkerService.java

```diff
@@ -143,6 +144,60 @@ private void cleanLoadFlowResultsAndPublishCancel(UUID resultUuid, String receiv
         notificationService.publishStop(resultUuid, receiver);
     }
 
+    private static LoadingLimits.TemporaryLimit handleEquipmentLimitViolation(Branch<?> branch, LimitViolationInfos violationInfo) {
+
+        Optional<com.powsybl.iidm.network.CurrentLimits> currentLimits = violationInfo.getSide().equals("ONE") ? branch.getCurrentLimits1() : branch.getCurrentLimits2();
+        if (violationInfo.getValue() < currentLimits.get().getPermanentLimit()) {
+            return null;
+        } else {
+            List<LoadingLimits.TemporaryLimit> temporaryLimits = currentLimits.get().getTemporaryLimits().stream().collect(Collectors.toList());
+            Optional<LoadingLimits.TemporaryLimit> nextTemporaryLimit = temporaryLimits.stream()
```

Comment:
there is no need for the tmp List
```suggestion
            Optional<LoadingLimits.TemporaryLimit> nextTemporaryLimit = currentLimits.get().getTemporaryLimits().stream()
```

Reply:
yes it's done